packages/shell-api/src/shell-internal-state.ts

```diff
@@ -0,0 +1,238 @@
+    const mongo = new Mongo(this);
+    this.currentCursor = null;
+    this.currentDb = mongo.getDB('test'); // TODO: set to CLI arg
+  }
+
+  async fetchConnectionInfo(): Promise<void> {
+    this.connectionInfo = await this.currentDb.mongo.serviceProvider.getConnectionInfo();
+    this.messageBus.emit('mongosh:connect', this.connectionInfo.extraInfo);
+  }
+
+  // TODO: do we need to close all Mongos ever made?
+  close(p): void {
+    try {
+      this.currentDb.mongo.serviceProvider.close(p);
+      // eslint-disable-next-line no-empty
+    } catch (e) {
+    }
+  }
+
+  use(db): any {
+    return this.currentDb.mongo.use(db);
+  }
+  show(arg): any {
+    return this.currentDb.mongo.show(arg);
+  }
+  async it(): Promise<any> {
+    if (!this.currentCursor) {
+      // TODO: warn here
+      return new CursorIterationResult();
+    }
+    return await this.currentCursor._it();
+  }
+  public help(): Help {
+    this.messageBus.emit('mongosh:help');
+    return new Help({
+      help: 'shell-api.help.description',
+      docs: 'https://docs.mongodb.com/manual/reference/method',
+      attr: [
+        {
+          name: 'use',
+          description: 'shell-api.help.help.use'
+        },
+        {
+          name: 'it',
+          description: 'shell-api.help.help.it'
+        },
+        {
+          name: 'show databases',
+          description: 'shell-api.help.help.show-databases'
+        },
+        {
+          name: 'show collections',
+          description: 'shell-api.help.help.show-collections'
+        },
+        {
+          name: '.exit',
+          description: 'shell-api.help.help.exit'
+        }
+      ]
+    });
+  }
+
+  /**
+   * Prepare a `contextObject` as global context and set it as context
+   * Add each attribute to the AsyncWriter also.
+   *
+   * The `contextObject` is prepared so that it can be used as global object
+   * for the repl evaluationi.
+   *
+   * @note The `contextObject` is mutated, it will retain all of its existing
+   * properties but also have the global shell api objects and functions.
+   *
+   * @param {Object} contextObject - contextObject an object used as global context.
+   */
+  setCtx(contextObject: any): void {
+    // Add API methods for VSCode and scripts
+    contextObject.use = this.use.bind(this);
+    contextObject.show = this.show.bind(this);
+    contextObject.it = this.it.bind(this);
+    contextObject.help = this.help.bind(this);
+    contextObject.toIterator = toIterator;
+    contextObject.print = async(arg) => {
+      if (arg.toReplString) {
+        console.log(await arg.toReplString());
+      } else {
+        console.log(arg);
+      }
+    };
+    contextObject.printjson = contextObject.print;
+    Object.assign(contextObject, ShellBson);
+    contextObject.rs = new ReplicaSet(this.currentDb.mongo);
+    contextObject.sh = new Shard(this.currentDb.mongo);
+
+    // Add global shell objects
+    this.asyncWriter.symbols.initializeApiObjects({
```

Comment:
This should be probably done in the constructor, or it does depend on the context.

Reply:
Lucy is trying to see if she can use the async-rewriter in java, but if she can't, then we will have to move the async-rewriter out of internal state entirely and put it in the shell-evaluator. I feel like if we're not setting the state in the same place (I was worried people would forget to add to both context and symbol table), might as well put it in a different package since java can't import babel and doesn't use shell-eval. That seems nicer anyway since rewriter has more to do with eval than it has to do with state imo.